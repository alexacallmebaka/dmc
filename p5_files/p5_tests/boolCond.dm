a:int;
b:int = 1;
c:bool = true;
f : () void {
	d:int;
}
C : class {
};
main : () void {
  a = 1;
  if (a == b) {
    if (c) {}
    else {
      while (C) {}
    }
  }
  if (false) {}
  while (false) {}
  while (a) {}
  while (f) {}
}