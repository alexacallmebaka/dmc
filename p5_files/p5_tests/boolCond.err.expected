FATAL [14,14]-[14,15]: Non-bool expression used as a condition
FATAL [19,10]-[19,11]: Non-bool expression used as a condition
FATAL [20,10]-[20,11]: Non-bool expression used as a condition
Type Analysis Failed
